window/src/os/x11/connection.rs

```diff
@@ -441,7 +454,17 @@ impl XConnection {
 
         let default_dpi = RefCell::new(compute_default_dpi(&xrm, &xsettings));
 
-        let conn = XConnection {
+        xcb_imdkit::ImeClient::set_logger(|msg| log::debug!("Ime: {}", msg));
```

Comment:
the debug log lines have extra newlines; this will eliminate them, but you may want to `trim_end` the string that you're passing to the logger callback in your crate so that this implementation detail is hidden :)

```suggestion
        xcb_imdkit::ImeClient::set_logger(|msg| log::debug!("Ime: {}", msg.trim_end()));
```